File: project/AccessControlSystem-Server/src/main/java/com/acs/model/DoorLock.java
package com.acs.model;

import javax.persistence.*;

@Entity
@Table(name = "acs_doorlock")
public class DoorLock {

    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private String name;

    @OneToOne(targetEntity = Scanner.class, fetch = FetchType.LAZY)
    private Scanner scanner;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Scanner getScanner() {
        return scanner;
    }

    public void setScanner(Scanner scanner) {
        this.scanner = scanner;
    }
}

File: project/AccessControlSystem-Client/src/app/components/models/DoorLock.ts
import {Scanner} from "./Scanner";

export class DoorLock {
  id: number;
  name: string;
  scanner: Scanner;


  constructor(id: number, name: string, scanner: Scanner) {
    this.id = id;
    this.name = name;
    this.scanner = scanner;
  }
}
